components/automate-cli/pkg/verifyserver/services/firewallservice/firewallservice.go

```diff
@@ -0,0 +1,155 @@
+			Passed:        false,
+			SuccessMsg:    "",
+			ErrorMsg:      fmt.Sprintf(constants.FIREWALL_ERROR_MESSAGE, req.DestinationServiceProtocol, req.DestinationNodeIP, req.DestinationServicePort, req.SourceNodeIP),
+			ResolutionMsg: fmt.Sprintf(constants.FIREWALL_RESOLUTION_MESSAGE, req.DestinationServicePort, req.DestinationNodeIP, req.SourceNodeIP),
+		}
+	}
+	fw.log.Debug("Response From port-reachable API: ", respBody)
+	return models.Checks{
+		Title:         constants.FIREWALL_TITLE,
+		Passed:        true,
+		SuccessMsg:    fmt.Sprintf(constants.FIREWALL_SUCCESS_MESSAGE, req.DestinationServiceProtocol, req.DestinationNodeIP, req.DestinationServicePort, req.SourceNodeIP),
+		ErrorMsg:      "",
+		ResolutionMsg: "",
+	}
+
+}
+
+func (fw *FirewallService) triggerRequest(req models.FirewallRequest, client *http.Client) (*models.Checks, error) {
+	fw.log.Debug("triggerRequest Called...")
+	url := fmt.Sprintf("http://%s:%s%s", req.SourceNodeIP, fw.port, constants.PORT_REACHABLE_API_PATH)
+	fw.log.Debug("URL: ", url)
+
+	destinationPort, err := strconv.Atoi(req.DestinationServicePort)
+	if err != nil {
+		return nil, errors.New("Failed to convert string ip into int: " + err.Error())
+	}
+	httpReqBody := models.PortReachableRequest{
+		DestinationNodeIp:              req.DestinationNodeIP,
+		DestinationNodePort:            destinationPort,
+		DestinationNodeServiceProtocol: req.DestinationServiceProtocol,
+		RootCA:                         req.RootCert,
+	}
+	httpReqBodyJSON, err := json.Marshal(httpReqBody)
+	if err != nil {
+		return nil, errors.New("Failed to Marshal: " + err.Error())
+	}
+	httpReq, err := http.NewRequest(http.MethodPost, url, bytes.NewBuffer(httpReqBodyJSON))
+	if err != nil {
+		return nil, errors.New("Error creating request: " + err.Error())
+	}
+	httpReq.Header.Set("Content-Type", "application/json")
+
+	resp, err := client.Do(httpReq)
+	if err != nil {
+		return nil, errors.New("Error sending request: " + err.Error())
+	}
+
+	return fw.getResultStructFromRespBody(resp.Body)
+}
+
+func (fw *FirewallService) getResultStructFromRespBody(respBody io.Reader) (*models.Checks, error) {
+	fw.log.Debug("getResultStructFromRespBody called...")
+	body, err := io.ReadAll(respBody)
+	if err != nil {
+		return nil, errors.New("Cannot able to read data from response body: " + err.Error())
+	}
+
+	// Converting API Response Body into Generic Response Struct.
+	apiRespStruct := response.ResponseBody{}
+	err = json.Unmarshal(body, &apiRespStruct)
+	if err != nil {
+		return nil, errors.New("Failed to Unmarshal: " + err.Error())
+	}
+
+	// If API(/port-reachable) is itself failing.
+	if apiRespStruct.Error != nil {
+		return nil, apiRespStruct.Error
+	}
+
+	// Converting interface into JSON encoding. apiResp.Result is a interface and for accessing the values we are converting that into json.
+	resultByte, err := json.Marshal(apiRespStruct.Result)
+	if err != nil {
+		return nil, errors.New("Failed to Marshal: " + err.Error())
```

Comment:
This message should be clear. Failed to Marshal/Unmarshal the response from Port reachability API